### DISCUSSION TITLE
internal routing options Brouter offline profile

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#316] [Q&A] [answered]
The new options generate funny and partially useful routing, but especially in villages or small cities it deviates from main roads and uses short and unnecessary detours through residential areas, although residential areas should be excluded.

These are my settings
<img src="https://github.com/devemux86/cruiser/assets/121371629/691f31e9-30c3-495a-b704-3fa01e611210" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/121371629/02adb28e-4247-4d56-b582-c57e96d2c82b" width="50%"/>

Even when using the new funny options, the "avoid residential roads" should get higher priority if driving through cities and villages which can't be avoided completely.

**Comment by devemux86** [comment, marked as answer]
@DolomitesTom thanks for the detailed settings.

---

The new BRouter options are still work in progress from its developers:
- https://github.com/abrensch/brouter/pull/614

_Advice: do not change anything in the "Other" category._

Info: "Maximum speed" affects the available roads in routing.

---

_Better use the detailed options for avoid road types in the main screen._

If in doubt about a route, please post the route here so we can check its tags.
- https://github.com/devemux86/cruiser/discussions/272

---

Avoid cities / villages doesn't work perfectly in all routers, that's for sure.
Best implementation is through setting no-go areas like BRouter website.

_This is already planned for Cruiser._ :slightly_smiling_face:

**Reply by DolomitesTom** [reply]
OK thanks, avoid cities taken out until further notice.

**Comment by bios4** [comment]
> Info: "Maximum speed" affects the available roads in routing.

How does this setting affect the planning in detail?

Setting it to a higher speed --> what is the expected outcome?
Setting it lower --> what can be expected then?

And what exactly is the very meaning of that setting at all?
- Is it my desired max. speed when travelling?
- Is it the speed limit of streets which are used to calculate the routes?
- ...?

**Reply by devemux86** [reply]
@bios4 

Lower max speed should use roads with lower max speed.

You can also test it in the [BRouter website](https://brouter.de/brouter-web/):
- https://github.com/devemux86/cruiser/discussions/272

**Reply by devemux86** [reply]
@bios4 

It could be seen as another way to avoid high speed roads.

**Comment by bios4** [comment]
> You can also test it in the [BRouter website](https://brouter.de/brouter-web/)

Is there a possibility to export the profile settings from my device in the proper format to be used on the BRouter website?
Because the new json format is not usable there...

**Reply by devemux86** [reply]
@bios4 these are 2 different things:
- **Cruiser** profiles contain the app settings (json extension)

- **BRouter** profiles contain routing algorithms (brf extension)
and we can test them in BRouter website.

BRouter customized driving profile that is used in Cruiser is at:
- https://github.com/devemux86/cruiser/discussions/272

**Reply by bios4** [reply]
OK...

So what I understand from that is that Cruiser always uses this "driving-cruiser.brf" profile, but what is contained in it is depending on the settings I apply within the app in "Settings > Routing > Options > BRouter (offline)".

If that is correct, how can I get the content of MY "driving-cruiser.brf" from MY device?

**Reply by devemux86** [reply]
@bios4 

Correctly, the `driving-cruiser` custom BRouter profile is used for car + motorcycle routing.
For bike and pedestrian routing, other BRouter internal profiles are used.

> how can I get the content of MY "driving-cruiser.brf" from MY device?

The profile can be downloaded in the mentioned guide:
- https://github.com/devemux86/cruiser/discussions/272

**Reply by bios4** [reply]
Emux - the link in above post leads me to the default brf file located on your GitHub.

But I'd like to know if (and "how", if at all) I can extract the contents of MY profile which was updated with MY settings in the app from MY device... ;-)

Am I missing something?

**Reply by devemux86** [reply]
@bios4 what do you want to do:
- Export your Cruiser profile with the app settings?
  - Use the profile manager menu to export the json file

- Or use the BRouter profile (as customized in Cruiser) in BRouter website?
  - Go to the guide to download the BRouter brf file: https://github.com/devemux86/cruiser/discussions/272

These are 2 separate things containing completely different data, as explained [above](https://github.com/devemux86/cruiser/discussions/316#discussioncomment-7048157).

**Comment by bios4** [comment]
> what do you want to do

I thought about using exactly MY Cruiser profile/BRouter profile/settings/whatever (which was adapted within the app on MY device and is NOT similar to the brf file one can download from above GitHub link) and "import" exactly these settings on the BRouter website to directly see how certain changes affect the routing.

And after tweaking the settings to my liking possibly "export" (copy) these settings from the BRouter website and bring it back in MY Cruiser app on MY device.

But it seems that this is not possible (right now at least).

**Reply by devemux86** [reply]
@bios4 things are much more simple:
- Go to the BRouter guide: https://github.com/devemux86/cruiser/discussions/272
- Download the BRouter `driving-cruiser` profile and follow the instructions
- Both Cruiser and BRouter website have the same routing options

**Reply by bios4** [reply]
Obviously we are writing past each other, so I give up.
Thanks anyway for your time!

**Reply by bios4** [reply]
After playing around yesterday evening I think the option in "Settings > Routing > Options > BRouter (offline) > Advanced > BRouter Profile > Open External Profile" is what I searched for...

As I understand it, with this option one can 1) get the default brf file (GitHub link above) > 2) load it on the BRouter website and tweak it as desired > 3) download/copy these settings as a (new) brf file > 4) copy it to the device where Cruiser is running and finally > 5) open it with above option to use it with the app.

Is that correct?

**Reply by devemux86** [reply]
@bios4 correctly! :+1: 

Sorry, I didn't understand what you described. :slightly_smiling_face: 
Indeed, this is the procedure if you want to load external BRouter profiles into Cruiser.

**Reply by bios4** [reply]
Are settings from an external profile overwriting any settings I set directly via the app's dialogs?

E.g. if i set "Max. speed" to 150 km/h in the app, and then load an external profile with vmax set to 100 km/h - which value is taken into account then?

**Reply by devemux86** [reply]
Routing options are selected in the app. BRouter profiles just contain the algorithms.
You can test them in BRouter website, then enable the same options in the app.

Loading external BRouter profiles in Cruiser only has meaning if you customize the algorithms.
Otherwise, if only want to test the options, just select the same options in the app and website.
